cartographer/sensor/compressed_point_cloud.cc

```diff
@@ -149,6 +149,17 @@ CompressedPointCloud::CompressedPointCloud(const std::vector<int32>& point_data,
                                            size_t num_points)
     : point_data_(point_data), num_points_(num_points) {}
 
+CompressedPointCloud::CompressedPointCloud(
+    const proto::CompressedPointCloud& proto) {
+  const size_t data_size = proto.point_data_size();
+
+  num_points_ = proto.num_points();
+  point_data_.reserve(proto.point_data_size());
+  for (int i = 0; i != data_size; ++i) {
+    point_data_.emplace_back(proto.point_data(i));
+  }
```

Comment:
As I said it would be nice to add a `CHECK()` that verifies that the proto contained valid data. Otherwise code might crash/misbehave at hard to debug places. At least we should have a `TODO` here. The point is not so much to check that we correctly serialized, but that we, e.g. did not read malformed data from disk.